### DISCUSSION TITLE
kibana dashboard in iframe redirects back to login page

### DISCUSSION THREAD

**Discussion by manoj30** [Security-Onion-Solutions/securityonion#4291] [Unsupported Versions] [answered]
Hi,
I am new to security onion. I wanted to to embed security onion home dashboard in iframe of another sample application.
```
<!DOCTYPE html>
  <body>
    <iframe src="https://10.0.2.15/kibana/goto/52e3644..."></iframe>
  </body>
</html>
```
(The app is hosted on another host so its cross origin). When I try to open the sample app it loads the login page.
However, after login with right credentials it again loads back the login page. 

The standalone kibana embed urls allow loading dashboard in iframe.

If the same kibana dashboard url is loaded from another tab, it works well as expected.

I tried to check the nginx config in `so-nginx`. Where it looks like the auth is forwarded to `so-kratos` for which uses some kratos framework.

It would be quite helpful if anyone could share the light on why the app in iframe only is showing this behavior.

**Comment by jertel** [comment, marked as answer]
Security Onion explicitly prevents itself, or any of the include apps hosted within, from being embedded in an IFRAME hosted from a different origin. This is to prevent bad actors from performing browser-based attacks on Security Onion users.

**Reply by manoj30** [reply]
Ok.. agreed.. But had a doubt in mind, If a Kibana dashboard has share iframe code functionality, then shouldn't it be allowed to be loaded in frame? (i.e. just allow the the kibana embed URLs)
